pkg/cmd/util/plug/plug.go

```diff
@@ -96,22 +107,31 @@ func NewLeased(leaser Leaser) *Leased {
 }
 
 // Stop releases the acquired lease
-func (l *Leased) Stop() {
+func (l *Leased) Stop(err error) {
 	l.leaser.Release()
-	l.Plug.Stop()
+	l.Plug.Stop(err)
 }
 
 // Run tries to acquire and hold a lease, invoking Start()
 // when the lease is held and invoking Stop() when the lease
-// is lost.
-func (l *Leased) Run() {
-	ch := make(chan struct{}, 1)
+// is lost. If the lease was lost gracefully, nil is returned.
+// If the lease was lost due to an error, the error is returned.
+func (l *Leased) Run() error {
+	ch := make(chan error, 1)
 	go l.leaser.AcquireAndHold(ch)
-	defer l.Stop()
+	var err error
+	defer l.Stop(err)
```

Comment:
`defer`'s function args are evaluated when we get to this line, not at the end of the function right?  Wouldn't this mean that `err` is always `nil`?